apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.BenchmarkingTool/Configuration.cs

```diff
@@ -29,29 +29,47 @@ internal sealed class Configuration
     internal string OutputFolder { get; }
     internal int UploadChunkSizeInBytes { get; }
     internal int DownloadChunkSizeInBytes { get; }
-    internal string Bucket { get; } = IdGenerator.FromDateTime(prefix: "benchmark-");
+    internal string BucketStorageClass { get; }
+    internal string BucketLocation { get; }
+    internal bool ObjectVersioningEnabled { get; }
+    internal string BucketName { get; } = IdGenerator.FromDateTime(prefix: "benchmark-");
     internal string ObjectName => "obj_random_data";
 
-    private Configuration(int numOfSamples, int dataSizeInBytes, string outputFolder, int uploadChunkSizeInBytes, int downloadChunkSizeInBytes)
+    private Configuration(
```

Comment:
One option to consider is having a *single* command line argument which refers to a JSON file containing the configuration. You could then just use Json.NET or similar to deserialize the whole configuration. It would also make it simple to maintain multiple configurations, as separate files, rather than have to mess about with different options on the command line.

Something for a separate PR probably, but worth thinking about.